rofl-appd/src/routes/tx.rs

```diff
@@ -53,21 +54,43 @@ pub enum Transaction {
     /// Ethereum transaction.
     #[serde(rename = "eth")]
     Eth {
-        gas_limit: u64,
-        #[serde_as(as = "serde_with::hex::Hex")]
+        gas_limit: QuantityU64,
+        #[serde(deserialize_with = "deserialize_hex_bytes")]
         to: Vec<u8>,
         value: TransactionValue,
-        #[serde_as(as = "serde_with::hex::Hex")]
+        #[serde(deserialize_with = "deserialize_hex_bytes")]
         data: Vec<u8>,
     },
 }
 
+/// A quantity that can be provided either as a JSON number or as a string
+/// (decimal or 0x-prefixed hex) and normalized into a `u64`.
+#[derive(Clone, Debug, serde::Deserialize)]
+#[serde(untagged)]
+pub enum QuantityU64 {
```

Comment:
I would rename this to something more specific, like `GasLimitInput`, since it's only used for that at the moment. Unless we see this being useful elsewhere as well.